test/e2e/lib/jest/environment.js

```diff
@@ -7,12 +7,11 @@ class JestEnvironmentE2E extends JestEnvironmentNode {
 	async handleTestEvent( event ) {
 		switch ( event.name ) {
 			case 'setup':
-				this.global.__CURRENT_TEST_NAME__ = null;
 				this.global.__CURRENT_TEST_FAILED__ = false;
 				break;
 
 			case 'test_start':
-				this.global.__CURRENT_TEST_NAME__ = event.test.name;
+				this.global.__CURRENT_SUITE_NAME__ = event.test.parent.name;
```

Comment:
How does this work with nested `describe` blocks? Should we traverse up all `parent`s and compose the name?

Reply:
It won't help if two different tests have the same internal `describe` block. They can cause collisions and miss some recording.

Ideally we should have the test name, but Jest doesn't expose it. The next best thing is concatenate all parent describe blocks

Reply:
That's an apt observation. There would be ways to work around it, I'll have to think about what would be the best (and maintainable) approach.

Reply:
Taking it back for some more thinking on my part.